### PR TITLE
add ie810

### PR DESCRIPTION
**Pull request by ChrisCookOC**

### CODE REVIEW

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/actions/ValidateErnsAction.scala [hunk 1]
```diff
@@ -23,23 +23,27 @@ import play.api.mvc.Results.Forbidden
...
+class ValidateErnsActionImpl @Inject()(messageService: MessageService)(
+  implicit val executionContext: ExecutionContext,
+  implicit val emcsUtils: EmcsUtils
+) extends ValidateErnsAction
+  with Logging {
   override def refine[A](request: ParsedXmlRequest[A]): Future[Either[Result, ValidatedXmlRequest[A]]] = {
 
-    val matchedErns: Set[String] = request.erns.intersect(request.ieMessage.getErns)
+    val matchedErns: Future[Set[String]] =
+      messageService.getErns(request.ieMessage).map(ernsFromMessage => request.erns.intersect(ernsFromMessage))
```
Comment (mauro-fanton) [inline app/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/actions/ValidateErnsAction.scala:39]: is there any reason that this return a feature? I do not see any reason that this is a feature as we only want to take the list of valid ern from an IEMessage

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/repository/MovementRepository.scala [hunk 2]
```diff
@@ -68,6 +68,10 @@ class MovementRepository @Inject()
       ))
       .toFuture()
   }
+
+  def getMovementByARC(arc: String): Future[Seq[Movement]] = {
```
Comment (mauro-fanton) [inline app/uk/gov/hmrc/excisemovementcontrolsystemapi/repository/MovementRepository.scala:72]: are there any test for this in the MovementRepositorySpec? I cannot see any.

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/services/MessageService.scala [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import uk.gov.hmrc.excisemovementcontrolsystemapi.models.MessageTypes.{IE801, IE810, IE815, IE818, IE837}
+import uk.gov.hmrc.excisemovementcontrolsystemapi.models.messages._
+import uk.gov.hmrc.excisemovementcontrolsystemapi.repository.MovementRepository
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class MessageService @Inject()(movementRepository: MovementRepository, implicit val executionContext: ExecutionContext) {
+
+  private def getErnsUsingArc(arc: String): Future[Set[String]] = {
```
Comment (mauro-fanton) [inline app/uk/gov/hmrc/excisemovementcontrolsystemapi/services/MessageService.scala:28]: what about somethiong like that which seem for functional? And may more readable as you just deal with one future only?

` movementRepository.getMovementByARC(arc).map {
      case Seq() => throw new RuntimeException(s"[MessageService] - Zero movements found for administrative reference code $arc")
      case head :: Nil => Set(Some(head.consignorId), head.consigneeId).flatten
      case _ => throw new RuntimeException(s"[MessageService] - Multiple movements found for administrative reference code $arc")
    }`

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/actions/ValidateErnsAction.scala (hunk 1 above)
Comment (ChrisCookOC) [reply app/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/actions/ValidateErnsAction.scala:39]: Yeah so for the IE810 we have to get the ERNs from the database so it returns a Future so the GetErns then returns a Future and then the intersection of the two sets has to be a Future - its futures all the way up

Comment (ChrisCookOC) [reply app/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/actions/ValidateErnsAction.scala:39]: already removed that in the last commit. (not sure why it took 2 hours for github to tell me you responded!)

**Review by mauro-fanton — APPROVED** [review]

**Review by mauro-fanton — APPROVED** [review]

**Review by mauro-fanton — APPROVED** [review]

**Review by mauro-fanton — APPROVED** [review]

**Review by mpbostock — APPROVED** [review]

**Pull request merged**